File: Uke1/løsningsforslag/oppgave4.py
from imageio import imread
import matplotlib.pyplot as plt
import numpy as np

f = imread('mona.png',as_gray=True)

for ii in range(1,8,2):
    bit = ii

    quantized = f//2**(8-bit) # Operasjonen // betyr at vi gjør heltallsdivisjon.

    plt.figure() # lag nytt vindu
    plt.title("bit = %d, antall verdier = %d"%(bit, np.max(quantized) + 1))
    plt.imshow(quantized,cmap='gray')

plt.show()

File: Uke1/løsningsforslag/oppgave4.m
f = double(imread('mona.png')); % double for at divisjon ikke avrunder verdier 

for ii = 1:2:8
    bit = ii; 
    f_requantized = floor(f./(2^(8-bit))); % floor kan brukes her til aa 
                                           % avrunde verdiene som om 
                                           % heltallsdivisjon har blitt utfoert 
                                           % paa dette bildet. 

    figure()
    imagesc(f_requantized); colormap('gray');
    title(sprintf('bit = %d, antall verdier = %d', bit, max(f_requantized(:))+1))
end